react/features/base/conference/actions.ts

```diff
@@ -1059,13 +1061,31 @@ export function redirect(vnode: string, focusJid: string, username: string) {
             .then(() => dispatch(conferenceWillInit()))
             .then(() => dispatch(connect()))
             .then(() => {
-
                 // Clear the gum pending state in case we have set it to pending since we are starting the
                 // conference without tracks.
                 dispatch(gumPending([ MEDIA_TYPE.AUDIO, MEDIA_TYPE.VIDEO ], IGUMPendingState.NONE));
 
                 // FIXME: Workaround for the web version. To be removed once we get rid of conference.js
                 if (typeof APP !== 'undefined') {
+                    if (!vnode) {
+                        const { enableMediaOnPromote = {} } = getState()['features/base/config'].visitors ?? {};
+                        const { audio = false, video = false } = enableMediaOnPromote;
+
+                        if (audio) {
+                            dispatch(setAudioMuted(false, true));
```

Comment:
What if the visitor initially joined with startWithAudioMuted/startWithVideoMuted or selected audio-only mode on join? Shouldn't we take those into consideration before unmuting them?

Reply:
hmm... We discussed we will ignore those for now. but I agree this is kind of an edge case. lets discuss offline with the team.

Reply:
actually I think audio only should work and unmute, I have to test it. But muting from the prejoin would be ignored. Otherwise if the config option is set we already discussed that the new properties will be higher priority.

Reply:
Will fix it

Reply:
Fixed!